File: rock_paper_scissors/java/RockPaperScissors.java
import java.util.Random;
import java.util.Scanner;

public class RockPaperScissors {

    public static void main(String[] args) {
        Random rand = new Random();
        Scanner scanner = new Scanner(System.in);
        System.out.println("ROCK, PAPER, SCISSORS");

        int wins = 0, losses = 0, ties = 0;

        while (true) {
            System.out.printf("%s Wins, %s Losses, %s Ties\n", wins, losses, ties);
            System.out.println("Enter your move: (r)ock, (p)aper, (s)cissors or (q)uit");
            String playerMove = scanner.next();

            if (playerMove.equals("q")) {
                break;
            }

            if (playerMove.equals("r")) {
                System.out.println("ROCK versus...");
            } else if (playerMove.equals("p")) {
                System.out.println("PAPER versus...");
            } else if (playerMove.equals("s")) {
                System.out.println("SCISSORS versus...");
            }

            String computerMove = "";
            int randNum = rand.nextInt(3);
            if (randNum == 0) {
                computerMove = "r";
                System.out.println("ROCK");
            } else if (randNum == 1) {
                computerMove = "p";
                System.out.println("PAPER");
            } else {
                computerMove = "s";
                System.out.println("SCISSORS");
            }

            if (playerMove.equals(computerMove)) {
                ties += 1;
                System.out.println("It is a tie!");
            } else if (playerMove.equals("r") && computerMove.equals("p")) {
                losses += 1;
                System.out.println("You lose!");
            } else if (playerMove.equals("p") && computerMove.equals("s")) {
                losses += 1;
                System.out.println("You lose!");
            } else if (playerMove.equals("s") && computerMove.equals("r")) {
                losses += 1;
                System.out.println("You lose!");
            } else if (playerMove.equals("r") && computerMove.equals("s")) {
                wins += 1;
                System.out.println("You win!");
            } else if (playerMove.equals("p") && computerMove.equals("r")) {
                wins += 1;
                System.out.println("You win!");
            } else if (playerMove.equals("s") && computerMove.equals("p")) {
                wins += 1;
                System.out.println("You win!");
            }
        }
    }
}

File: rock_paper_scissors/python/rock_paper_scissors.py
import sys, random

print('ROCK, PAPER, SCISSORS')

wins = 0
losses = 0
ties = 0

setOfChoices = ['r', 'p', 's']

while True:
    while True:
        print('%s Wins, %s Losses, %s Ties' % (wins, losses, ties))
        print('Enter your playerMove: (r)ock (p)aper (s)cissors or (q)uit')
        playerMove = input()

        if playerMove.lower().startswith('q'):
            sys.exit()
        elif playerMove == 'r' or playerMove == 'p' or playerMove == 's':
            break

    if playerMove == 'r':
        print('ROCK versus...')
    elif playerMove == 'p':
        print('PAPER versus...')
    elif playerMove == 's':
        print('SCISSORS versus...')

    computerMove = setOfChoices[random.randint(0, 2)]
    if computerMove == 'r':
        print('ROCK')
    elif computerMove == 'p':
        print('PAPER')
    elif computerMove == 's':
        print('SCISSORS')

    if playerMove == computerMove:
        ties += 1
        print('It is a tie!')
    elif playerMove == 'r' and computerMove == 'p':
        losses += 1
        print('You lose!')
    elif playerMove == 'p' and computerMove == 's':
        losses += 1
        print('You lose!')
    elif playerMove == 's' and computerMove == 'r':
        losses += 1
        print('You lose!')
    elif playerMove == 'r' and computerMove == 's':
        wins += 1
        print('You win!')
    elif playerMove == 'p' and computerMove == 'r':
        wins += 1
        print('You win!')
    elif playerMove == 's' and computerMove == 'p':
        wins += 1
        print('You win!')